src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -56,67 +56,72 @@
 
 namespace
 {
+constexpr int maxTouchPoints = 10; // Current Windows API limitation
+
+// Define constants for tablet API to avoid pulling in an extra Tablet PC header
+const char*              tabletServiceProperty     = "MicrosoftTabletPenServiceProperty";
+const unsigned long long tabletDisablePressAndHold = 0x00000001ULL;
+
 unsigned int               windowCount      = 0; // Windows owned by SFML
 unsigned int               handleCount      = 0; // All window handles
 const wchar_t*             className        = L"SFML_Window";
 sf::priv::WindowImplWin32* fullscreenWindow = nullptr;
+DWORD                      touchIDs[maxTouchPoints];
+POINT                      touches[maxTouchPoints];
 
-const GUID GUID_DEVINTERFACE_HID = {0x4d1e55b2, 0xf16f, 0x11cf, {0x88, 0xcb, 0x00, 0x11, 0x11, 0x00, 0x00, 0x30}};
-
-void setProcessDpiAware()
+// Convert a hardware dependent ID to a 0 based index we can use
+int getTouchID(DWORD id)
```

Comment:
It looks like `-1` is being used as a sentinel value to represent a null state. Does the Windows API expect that? Could we instead return `std::optional<SomeIntegerType>` to better model what's going on?

Reply:
Another question, doesn't our new naming convention require `getTouchId` instead of `getTouchID`?
Not sure if this is enforced by CI, but would at least be consistent.

Reply:
And yes, I think `std::optional` would fit much better than the magic constant `-1`.

See also this occurrence, here we could just simply reset the optional:

https://github.com/SFML/SFML/blob/d313636448693173dba0798346f49df830eabc0a/src/SFML/Window/Win32/WindowImplWin32.cpp#L1074-L1075

Reply:
> It looks like `-1` is being used as a sentinel value to represent a null state.
It is, since Windows doesn't inherently define a null state and `0` might be a valid entry.